src/main/java/fi/fmi/avi/archiver/message/populator/MessageAerodromeValidator.java

```diff
@@ -0,0 +1,57 @@
+package fi.fmi.avi.archiver.message.populator;
+
+import fi.fmi.avi.archiver.file.InputAviationMessage;
+import fi.fmi.avi.archiver.file.InputBulletinHeading;
+import fi.fmi.avi.archiver.message.ArchiveAviationMessage;
+import fi.fmi.avi.archiver.message.ProcessingResult;
+import fi.fmi.avi.model.GenericAviationWeatherMessage;
+import fi.fmi.avi.model.bulletin.BulletinHeading;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static fi.fmi.avi.archiver.message.populator.MessagePopulatorHelper.DEFAULT_BULLETIN_HEADING_SOURCES;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * When the bulletin location indicator matches the given pattern, validate that the message aerodrome matches the given pattern.
+ */
+public class MessageAerodromeValidator implements MessagePopulator {
```

Comment:
I'd like the name of this class to describe that this validator restricts allowed stations for a configured sender to only configured ones. But I came up with no better suggestions so far. Any ideas?

Reply:
`StationIcaoCodeAuthorizer`?

Reply:
'Authorizer' sounds good. `OriginatorStationsAuthorizer` or `StationsOfOriginatorAuthorizer`? (I picked 'originator' from the description of `CCCC` in WMO GTS Manual)

Reply:
Or `AuthorizedOriginatorStationChecker`?